flake.lock

```diff
@@ -0,0 +1,49 @@
+{
```

Comment:
So this file pins all the revisions for things mentioned in the flake.nix?

Reply:
Yes. This is the main purpose of nix flakes. Separate specification of "channel" (defined as inputs in flake.nix) and "state" (locked versions picked up by the current state of this flake by flake.lock).
flake.nix is flexible enough for specifying git branch level channel or pinned commit. In any case, flake.lock should be made for a full state pinning.

Reply:
`nix flake update` will update flake.lock (and check in the file in git tree by default). Before the update, the inputs including nixpkgs will be the same regardless of other changes.